File: day01/part1.go
package main

import "io/ioutil"
import "fmt"

func main() {
	input, _ := ioutil.ReadFile("input")
	sum := 0
	for i := 0; i < len(input); i++ {
		switch input[i] {
		case '(':
			sum++
		case ')':
			sum--
		}
	}
	fmt.Println(sum)
}

File: day01/part1.rs
use std::io::{self, Read};

fn main() {
    let mut input = String::new();
    let stdin = io::stdin();
    stdin.lock().read_to_string(&mut input).expect("Could not read stdin");
    let output: i32 = input.chars()
        .map(|item| match item {
            '(' => 1,
            ')' => -1,
            _ => 0
        })
        .sum();
    println!("{}",output);
}
